docs/how-to-guides/add-dependencies.rst

```diff
@@ -0,0 +1,13 @@
+.. _how-to-guides-add-dependencies:
+
+*******************
+Adding dependencies
```

Comment:
Is this page needed? At least it should start by stating that this is all standardized, and nothing is specific to `meson-python`. It looks like the Python Packaging User Guide has no good section on this - maybe we should fix that (after finishing up this PR) so we can link to it for more details.

Reply:
I didn't want to get into it in the tutorial, but wanted to have somewhere to point to quickly learn this. I'd say keep it for now, try to fix the Packaging User Guide, and then update this based on the outcome.